### ISSUE TITLE
The Hive doesn't get their Free Carrier slot

### ISSUE THREAD

**Issue by EpicJoston** [RMC-14/RMC-14#4492]
## Description
The Hive is supposed to get a free Hivelord **and** a free Carrier without cramping the Tier 2 limits. How it currently works is the Hive gets a free Hivelord **or** a free Carrier, and the other _free_ one takes up space in the limited Tier 2 evolutions.

**Comment by Vermidia** [comment]
This is actually not the main problem, the main problem is if you lose a bunch of xenos since the freeSlots aren't "filled" and are instead subtracted from the actual ratio of t2s and t3s, if a bunch of xenos die and you're already past the limit for t2s you can no longer evolve to hivelord/carrier even with free slots

**Comment by EpicJoston** [comment]
Xenos dying and making the tier 2s and tier 3s wonky is something I suspected but had not gotten around to testing just yet. Although that sounds like a potentially seperate issue.

My test was I generated enough Xenos in the Hive to support 2 Tier 3s, 3 Tier 2s, and 4 Tier 1s. With none of them dying, my hive was: Queen, Crusher, Crusher, Warrior, Warrior, Warrior, Drone, Drone, Defender, Defender. I was only able to evolve one of the drones at a time, not both despite there being a free slot for each Drone evo.
